app.py

```diff
@@ -1,17 +1,44 @@
 import os
-from flask import Flask, request, render_template, redirect
+from flask import Flask, request, render_template, redirect, flash
 from lib.database_connection import get_flask_database_connection
+<<<<<<< HEAD
+import re
+=======
+>>>>>>> b927c6e62dcabddfba1a5c63b528721febbb0e0e
 from lib.space_repository import SpaceRepository
 
 # Create a new Flask app
 app = Flask(__name__)
+app.secret_key = 'your_secret_key_here'
 
 # == Your Routes Here ==
 
 # GET /index
 # Returns the homepage
 # Try it:
 #   ; open http://localhost:5000/
+
+def is_valid_email(email):
+    return "@" in email and "." in email
+
+def check_valid_password(password):
+    validation_messages = []
+    if len(password) < 8:
+        validation_messages.append('Password must be at least 8 characters.')
+    if not any(char.isupper() for char in password) or not any(char.islower() for char in password):
+        validation_messages.append('Password must contain uppercase and lowercase characters.')
+    if not any(char.isdigit() for char in password):
+        validation_messages.append('Password must contain at least 1 number.')
+    if not re.search(r'[!@#$%^&*(),.?":{}|<>]', password):
+        validation_messages.append('Password must contain at least 1 symbol.')
+    if not validation_messages:
+        return True
+    else:
+        return validation_messages
```

Comment:
Nice work on this validation logic and giving clear messages to user if they don't meet the specified conditions :).